src/sql/src/plan/statement.rs

```diff
@@ -726,9 +722,16 @@ fn handle_create_sink(
             )?
         }
         Connector::Kinesis { .. } => unsupported!("Kinesis sinks"),
-        Connector::AvroOcf { path } => avro_ocf_sink_builder(format, with_options, path, suffix)?,
+        Connector::AvroOcf { path } => avro_ocf_sink_builder(format, path, suffix)?,
     };
 
+    if !with_options.is_empty() {
+        bail!(
+            "unexpected parameters for CREATE SINK: {}",
```

Comment:
we _might_ want to change this error string to be a bit more descriptive? someone might use `consistency` with file sinks or something, that options not unexpected for `CREATE SINK` generally, just for file sinks. not sure how to best balance user friendliness vs duplicating this error handling across N functions

either way I still think this is an improvement over the old code